app/settings.go

```diff
@@ -38,6 +40,21 @@ type settings struct {
 	schema SettingsSchema
 }
 
+func (s *settings) PrimaryColor() string {
+	s.propertyLock.RLock()
+	defer s.propertyLock.RUnlock()
+	return s.schema.PrimaryColor
+}
+
+// OverrideTheme allows the settings app to temporarily preview different theme details.
+// Please make sure that you remember the original settings and call this again to revert the change.
+func (s *settings) OverrideTheme(theme fyne.Theme, name string) {
```

Comment:
I wondered about that but it overrides the theme as well

Reply:
Would it be possible to override the theme once, and then switch out the chosen color inside that theme? In that case your original naming would work and `name string` param can be dropped

Reply:
If you have a custom theme that supports mutating the primary colour this can be done.
However we do not currently have an API that supports extending a theme to switch the colour, so it seemed like doing it here was the only choice.
Open to better ideas

Reply:
I can't think of any better approaches right now, but I'm not sure this is a good addition to the public API. Though I guess `settings` is unexported, can it be `overrideTheme`?

Reply:
If it was unexported it would not be possible to call it using the temporary interface at:
https://github.com/fyne-io/fyne/pull/1292/files#diff-fc4b8de1d5fbe30aec3d3d9d22e86237R102.

As you say this is not exported in Settings API, but it can be used by other developers if they dig deep enough.
This is a special case as we want to apply the theme temporarily without updating the whole application.

The only other approach I can think of is to spwan another process, save the output to an image and load the image file in (becuse we cannot run multiple apps in one runtime, and because the widgets use "current theme" to render).

Reply:
Ah okay, disregard